pkg/activator/certificate/cache_test.go

```diff
@@ -141,7 +141,7 @@ func TestReconcile(t *testing.T) {
 		cr.certificatesMux.RLock()
 		defer cr.certificatesMux.RUnlock()
 		cert, err := cr.GetCertificate(nil)
```

Comment:
I thought if there is no error cert should be not nil?

Reply:
Not an error, but cr.certificate can be nil when called to fast (which is what happens in that test I think). With the if, it will just do another poll iteration, without it, it panics inside the `DeepEqual`.

Reply:
Ah, and to answer the question, nil is not really an error state I think (Or I assume that was the idea back then).

Reply:
Could we just have `func (cr *CertCache) GetCertificate(_ *tls.ClientHelloInfo) (*tls.Certificate) {` as well? It does not seem to implement an interface or something.